platform/src/components/aws/ssr-site.ts

```diff
@@ -1030,6 +1233,153 @@ async function handler(event) {
+        if (
+          protection.mode !== "oac-with-edge-signing" ||
+          ("edgeFunction" in protection && protection.edgeFunction?.arn)
+        ) {
+          return undefined;
+        }
+
+        const edgeConfig =
+          "edgeFunction" in protection ? protection.edgeFunction : {};
+        const memory = edgeConfig?.memory ? toMBs(edgeConfig.memory) : 128;
+        const timeout = edgeConfig?.timeout ? toSeconds(edgeConfig.timeout) : 5;
+
+        // Create IAM role for Lambda@Edge using SST transform pattern
+        const edgeRole = new aws.iam.Role(
+          ...transform(
+            undefined,
+            `${name}EdgeFunctionRole`,
+            {
+              assumeRolePolicy: JSON.stringify({
+                Version: "2012-10-17",
+                Statement: [
+                  {
+                    Action: "sts:AssumeRole",
+                    Effect: "Allow",
+                    Principal: {
+                      Service: [
+                        "lambda.amazonaws.com",
+                        "edgelambda.amazonaws.com",
+                      ],
+                    },
+                  },
+                ],
+              }),
+              managedPolicyArns: [
+                "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+              ],
+            },
+            { parent: self },
+          ),
+        );
+
+        // Create the Lambda@Edge function using SST transform pattern
+        const edgeFunction = new aws.lambda.Function(
+          ...transform(
+            undefined,
+            `${name}EdgeFunction`,
+            {
+              runtime: "nodejs22.x",
+              handler: "index.handler",
+              role: edgeRole.arn,
+              code: new pulumi.asset.AssetArchive({
+                "index.js": new pulumi.asset.StringAsset(`
+// Lambda@Edge function for automatic SHA256 header signing
+// This function adds the required x-amz-content-sha256 header for POST/PUT/PATCH requests
+// going to Lambda function URLs with Origin Access Control enabled.
+
+const crypto = require('crypto');
+
+exports.handler = async (event) => {
+  const request = event.Records[0].cf.request;
+  
+  // Only process requests that need SHA256 signing (methods with body)
+  if (!['POST', 'PUT', 'PATCH', 'DELETE'].includes(request.method)) {
+    return request;
+  }
+  
+  try {
+    // Get the request body
+    let bodyString = '';
+    
+    if (request.body && request.body.data) {
+      // Lambda@Edge provides body as base64-encoded string
+      if (request.body.encoding === 'base64') {
+        // Decode base64 to get the actual body content
+        bodyString = Buffer.from(request.body.data, 'base64').toString('utf8');
+      } else {
+        // If not base64 encoded, use as-is
+        bodyString = request.body.data;
+      }
+    }
+    
+    // Compute SHA256 hash of the body
+    const hash = crypto.createHash('sha256').update(bodyString, 'utf8').digest('hex');
```

Comment:
the below code fixed the file upload issue 

still, lambda@edge has a 1mb payload cap, so big uploads return the failed to verify hash error

maybe we can return a different error or put a heads-up in the docs

```suggestion
    // Get the request body as raw bytes (never convert to UTF-8 string)
    const bodyBuffer = request.body?.data
      ? (request.body.encoding === 'base64'
          ? Buffer.from(request.body.data, 'base64')
          : Buffer.from(request.body.data))
      : Buffer.alloc(0);

    // Compute SHA256 hash of the raw bytes
    const hash = crypto.createHash('sha256').update(bodyBuffer).digest('hex');
```

Reply:
Nice catch, I had not tested the file upload, shame on me.
I've added your suggested code, a note about the file size limit and also a better error type: https://github.com/sst/sst/pull/5581/commits/0791c643df9f7a887cb0a36fc40ca46a9d814bfa

Reply:
don't worry. your pr is one of the few that worked first try haha